src/NuGet.Clients/VsExtension/VsEvents/OnBuildPackageRestorer.cs

```diff
@@ -490,6 +558,7 @@ var waitDialogFactory
                     return;
                 }
 
+                _packageCount += packages.Count;
```

Comment:
>_packageCount [](start = 16, length = 13)

Presumable we need instance state here because these methods are void. Could we avoid this by making RestorePackagesOrCheckForMissingPackagesAsync (and the other restore method) return a RestoreTelemetryData (naming?) private type?